### PR TITLE
moving Mapstyle selector to Option menu  with select2 and rebased from  #2022 

### PR DESCRIPTION
**Pull request by callmebackdraft**
*added dropdown menu to Option menu with map styles in select2 style
*made the dropdown read localstorage so it reads the last used style
*made it compatible with local with .json files
*removed mapstyle selector from map itself

More screen real estate especially on mobile
 also responds to #1674

## Motivation and Context
real estate upgrade especially on small screens (see issue: #1674)

## How Has This Been Tested?
after editing loaded page on:
 Microsoft Edge 
 Chrome ( Win10)
 IPhone Safari
 IPhone Chrome

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: templates/map.html [hunk 1]
```diff
@@ -145,6 +145,12 @@
 					</label>
 				</div>
 			</div>
+			<div class="form-control">
+				<hr width="100%">
+				<label for="map-style">
+					<h3>Style</h3>
+					<select id="map-style" style="min-width:100%;"></select>
```
Comment (brycedorn) [inline templates/map.html]: needs `</label>`

File: static/map.js [hunk 2]
```diff
@@ -1054,6 +1054,47 @@ $(function () {
...
+		// recall saved mapstyle
+		$selectStyle.val(Store.get('map_style')).trigger("change");
+		
+	});
+	
+	
+	
+});
+
+//var changeStyle = document.getElementById('map-style')
```
Comment (brycedorn) [inline static/map.js]: rm commented out code? or do you want to ship w/ this

**Comment by callmebackdraft** [comment]
@brycedorn the commented out code has been removed ( small oversight ) as with the missing </label>
added it now, thanks for pointing them out.

**Comment by callmebackdraft** [comment]
Rebased PR to be compatible withe all the commits done the past day

**Comment by invisiblek** [comment]
please rebase and clean up the excessive whitespace as well as use the spaces vs tab convention already in these files

**Comment by callmebackdraft** [comment]
Refactored/rebased PR #2667 also edited the whitespace and spaces vs tab convention @invisiblek

**Pull request closed without merging**